### PR TITLE
[DLS-9603]: updated SolePropreioterTrading name to include index, mode and cache …

### PR DESCRIPTION
**Pull request by BackRowOla**
…form data

### CODE REVIEW

File: app/uk/gov/hmrc/fhregistrationfrontend/controllers/BusinessPartnersTradingNameController.scala [hunk 1]
```diff
@@ -16,53 +16,64 @@
...
 import uk.gov.hmrc.fhregistrationfrontend.config.FrontendAppConfig
 import uk.gov.hmrc.fhregistrationfrontend.forms.definitions.TradingNameForm.tradingNameForm
+import uk.gov.hmrc.fhregistrationfrontend.pages.businessPartners.SoleProprietorsTradingNamePage
+import uk.gov.hmrc.fhregistrationfrontend.repositories.SessionRepository
 import uk.gov.hmrc.fhregistrationfrontend.views.Views
 
 import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
 
 class BusinessPartnersTradingNameController @Inject()(
```
Comment (gjstirling) [inline app/uk/gov/hmrc/fhregistrationfrontend/controllers/BusinessPartnersTradingNameController.scala:31]: Should this be renamed ? 

BusinessPartnersTradingNameController => BusinessPartnersSoleProprietorsTradingNameController

File: it/uk/gov/hmrc/fhregistrationfrontend/controllers/BusinessPartnersTradingNameControllerISpec.scala [hunk 2]
```diff
@@ -1,114 +1,205 @@
...
+              val page = Jsoup.parse(res.body)
+              page.title must include("Does the partner’s business use a trading name that is different from its registered name?")
+              page.getElementsByTag("h1").text must include("Does Test User’s business use a trading name that is different from its registered name?")
+              val tradingNameValueField = page.getElementById("tradingName_value")
+              tradingNameValueField.hasAttr("value") mustBe false
+            }
+          }
+
+        }
+
```
Comment (gjstirling) [inline it/uk/gov/hmrc/fhregistrationfrontend/controllers/BusinessPartnersTradingNameControllerISpec.scala:70]: Add test here for loading form data from cache, similar to above but asserting value is true and textbox has trading name inside.

File: it/uk/gov/hmrc/fhregistrationfrontend/controllers/BusinessPartnersTradingNameControllerISpec.scala [hunk 3]
```diff
@@ -1,114 +1,205 @@
...
-  }
+    }
 
-  s"POST $route" when {
+    s"POST ${route(mode)}" when {
+      "form with no errors" should {
+        Map(
+          "override" -> List(userAnswersWithPageData(noTradingName), userAnswersWithPageData(tradingName)),
+          "add" -> List(emptyUserAnswers)
+        ).foreach { case (uaAction, userAnswers) =>
```
Comment (gjstirling) [inline it/uk/gov/hmrc/fhregistrationfrontend/controllers/BusinessPartnersTradingNameControllerISpec.scala:92]: uaAction => userAnswersAction

**Review by gjstirling — CHANGES_REQUESTED** [review]

File: it/uk/gov/hmrc/fhregistrationfrontend/controllers/BusinessPartnersTradingNameControllerISpec.scala (hunk 2 above)
Comment (BackRowOla) [reply it/uk/gov/hmrc/fhregistrationfrontend/controllers/BusinessPartnersTradingNameControllerISpec.scala:70]: The radio buttons values don't actually change, they are fixed to true for the 'Yes' radio option and false for the 'No' option. However, I can add a test to check the value for the textbox.

**Review by gjstirling — APPROVED** [review]

**Pull request merged**